lib/jobs/dell-wsman-get-systemcomponents.js

```diff
@@ -62,25 +70,40 @@ function DellWsmanGetSystemConfigComponentsFactory(
     };
 
     DellWsmanGetSystemConfigComponentsJob.prototype._handleSyncResponse = function(result) {
+        var self = this;
         if (typeof result !== "undefined" && result !== null) {
-            return waterline.catalogs.create({
-                node: this.nodeId,
-                source: result.source,
-                data: result.data
+            return waterline.catalogs.findLatestCatalogOfSource(self.nodeId, result.source)
+            .then(function(catalog){
+                if (_.isEmpty(catalog)) {
+                    logger.debug("handleAsyncResponse: Catalog (" + result.source + ") not found.  Creating...");
+                    return waterline.catalogs.create({
+                        node: self.nodeId,
+                        source: result.source,
+                        data: result.data
+                    });
+                } else {
+                    logger.debug("handleAsyncResponse: Catalog (" + result.source + ") found!  Updating...");
+                    return waterline.catalogs.updateByIdentifier(catalog.id, {data: result.data});
+                }
             });
         } else {
             logger.info('No Result from getComponents to create catalog');
+            return Promise.resolve();
         }
     };
 
     DellWsmanGetSystemConfigComponentsJob.prototype.getComponents = function(obm) {
-
-        if (!validator.isIP(this.options.serverIP) || !validator.isIP(this.options.shareAddress)) {
-            throw new Error('Invalid ServerIP/ShareAddress');
-        }
         var self = this;
-        this.options.serverUsername = obm.config.user;
-        this.options.serverPassword = encryption.decrypt(obm.config.password);
+        if (!self.options.serverIP || !self.options.serverUsername || ! self.options.serverPassword) {
```

Comment:
Confusion: if the logic shall be that if server information is inputted via payload, then there is no need to check OBM. Besides, such check point shall be placed in _initJob method?